### ISSUE TITLE
documentation: true outputs to stdout when -v0 is enabled

### ISSUE THREAD

**Issue by mpickering** [haskell/cabal#5991]
`-v0` should never output anything to stdout. cabal-install should be silent in this module of operation.

If you enable `documentation: true` in a cabal file then a lot of junk from haddock is output.

Perhaps passing `--no-warnings` to haddock is sufficient.

**Comment by harpocrates** [comment]
I don't think `--no-warnings` will be enough. Even besides that, the Haddock codebase still has a bunch of unguarded `putStrLn`'s for emitting various messages.

Once all that fixed, I think it should be enough to make sure that `-v0` in Cabal implies `-v0` in Haddock.

**Comment by 23Skidoo** [comment]
@harpocrates Is there a corresponding upstream ticket?

**Comment by harpocrates** [comment]
Nope, but there probably should be. There’s some nice beginner-friendly refactor work there to do.
